Improve the third “Typing React State” exercise

chasenlehara opened the pull request:
This:

- Rewords the comments to prompt participants to finish the exercise
- Replaces the useRef examples with something more realistic (input refs)
- Refactors all the JSX to use semantic HTML

Exercise:

- [Problem](https://codesandbox.io/p/sandbox/github/bitovi/trainings/tree/typing-react-state-exercise-3/typing-react-state/Exercise3/problem?file=src/App.tsx)
- [Solution](https://codesandbox.io/p/sandbox/github/bitovi/trainings/tree/typing-react-state-exercise-3/typing-react-state/Exercise3/solution?file=src/App.tsx)

CMLabay reviewed (approved):
Looks great!

avinavdixit25 commented on typing-react-state/Exercise3/problem/src/App.tsx:
```diff
@@ -1,56 +1,56 @@
+Exercise:
+
+1) Add types to all of the local state variables.
+2) Bonus: what happens if we replace useRef with useState?
```
The changes are great but I really wanted to go over 2 things that's missing here. Firstly, I wanted to avoid using `useState` to get the "Full Name" to show how `useRef` doesn't cause a re-render when `ref` is changed. Right now we use `useState` to create "Full Name" and so I think the 2nd part of the question now has no purpose. 
Secondly, I wanted to show the conditions under which a `ref` is immutable and how we can change it. This is why I had added 
```
const firstNameRef = useRef<string>(null);
```
This will force the participants to change the `useRef` definition in order to make it mutable. Can we do something about these things?

chasenlehara replied:
Thanks Avinav, this is helpful feedback. We talked about this during the dry run and I _think_ we wanted to focus on real-world cases where you would need this. This example felt contrived, so I could use some help coming up with a real-world use case to show this.

@CMLabay, I want to make sure this exercise includes what we cover in the training.

chasenlehara replied:
@avinavdixit25 @CMLabay Could you please take a look at the updated code? I’ve added a `previousFullName` ref that I think covers this feedback.

CMLabay replied:
@chasenlehara It looks like this covers it

chasenlehara committed:
Improve the third “Typing React State” exercise

This:

- Rewords the comments to prompt participants to finish the exercise
- Replaces the useRef examples with something more realistic (input refs)
- Refactors all the JSX to use semantic HTML
